dags/gmail_data_import_pipeline.py

```diff
@@ -168,6 +170,47 @@ def gmail_link_message_thread_ids_etl(**kwargs):
         LOGGER.info('Loaded table: %s', table_name)
 
 
+def gmail_history_details_etl(**kwargs):
+    data_config = data_config_from_xcom(kwargs)
+    user_id = get_gmail_user_id()
+    table_name = data_config.table_name_history_details
+    dataset_name = data_config.dataset
+    project_name = data_config.project_name
+
+    start_id = get_max_history_id_from_bq(
+                            project_name=data_config.project_name,
+                            dataset=dataset_name,
+                            column_name='historyId',
+                            table_name=data_config.table_name_thread_details
+                        )
+
+    with TemporaryDirectory() as tmp_dir:
```

Comment:
This block now appears four times (and passes the [rule of three](https://blog.codinghorror.com/rule-of-three/)). It seems to be worth refactoring or were you planning to do that as a separate PR. All that seems to be different is the DataFrame and the target BigQuery table. In `data-science-dags` I chose to follow Pandas' `to_gbq` interface. But could be any interface.

Reply:
have created a new issue

Reply:
Just adding the link to the issue: https://github.com/elifesciences/data-hub-issues/issues/161